lib/authn.js

```diff
@@ -0,0 +1,86 @@
+const config = require('../lib/config');
+const csrf = require('../lib/csrf');
+const sqldb = require('@prairielearn/postgres');
+
+const sql = sqldb.loadSqlEquiv(__filename);
+
+// authnParams { authUid, authName, authUin }
+//    or
+// authnParams { user_id } so that we skip the users_select_or_insert call
+//                         and setup the session without making the user
+//
+// authn_provider_name: short name for the provider
+//
+// See optionsDefault:
+const optionsDefault = {
+  pl_authn_cookie: true,
+  redirect: false,
+  locals_authn: true,
+};
+
+module.exports.load_user_profile = async (
```

Comment:
Let's add some JSDoc types to this function. You can add `// @ts-check` to the top of this file (and ideally the other files you modified) to get TypeScript to check things for you. Here's a starting point (this is using my other suggestion of a single `options` argument at the end):

```js
/**
 * @typedef {Object} LoadUserOptions
 * @property {'Google' | 'SAML'} provider
 */

/**
 * @param {import('express').Request} req
 * @param {import('express').Response} res
 * @param {LoadUserOptions} options
 */
module.exports.loadUser = async (
```

You'll need to add any other necessary properties (and additional `provider` types) to the `LoadUserOptions` type.